Content.Server/Kitchen/EntitySystems/SharpSystem.cs

```diff
@@ -42,28 +43,28 @@ private void OnAfterInteract(EntityUid uid, SharpComponent component, AfterInter
         if (args.Target is null || !args.CanReach)
             return;
 
-        TryStartButcherDoafter(uid, args.Target.Value, args.User);
+        args.Handled = TryStartButcherDoAfter(uid, args.Target.Value, args.User);
     }
 
-    private void TryStartButcherDoafter(EntityUid knife, EntityUid target, EntityUid user)
+    private bool TryStartButcherDoAfter(EntityUid knife, EntityUid target, EntityUid user)
     {
         if (!TryComp<ButcherableComponent>(target, out var butcher))
-            return;
+            return false;
 
         if (!TryComp<SharpComponent>(knife, out var sharp))
-            return;
+            return false;
 
         if (butcher.Type != ButcheringType.Knife)
         {
             _popupSystem.PopupEntity(Loc.GetString("butcherable-different-tool", ("target", target)), knife, user);
-            return;
+            return false;
```

Comment:
should it not handle the event if it made a popup to prevent other systems also handling it and making duplicate popups

Reply:
True, I'll move it down to group them as well.